### PR TITLE
Update to latest cadvisor and containerd (v1.3.3)

### PR DESCRIPTION
**Pull request by dims**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
We use containerd 1.3.x in [cluster-api](https://github.com/kubernetes-sigs/image-builder/blob/master/images/capi/packer/config/containerd.json#L2) / [kind](https://github.com/kubernetes-sigs/kind/blob/master/images/base/Dockerfile#L26) already, it's time to move up to newer dependencies.

Need reviews for/from:

sig-windows:
- [x] github.com/Microsoft/go-winio
- [x] github.com/Microsoft/hcsshim
- [x] github.com/konsorten/go-windows-terminal-sequences

sig-node:
- [x] github.com/containerd/cgroups
- [x] github.com/containerd/console
- [x] github.com/containerd/containerd
- [x] github.com/containerd/ttrpc
- [x] github.com/godbus/dbus
- [x] github.com/google/cadvisor
- [x] github.com/mistifyio/go-zfs
- [ ] github.com/opencontainers/runc
- [x] github.com/opencontainers/runtime-spec
- [ ] github.com/opencontainers/selinux

sig-apimachinery:
- [x] github.com/gogo/protobuf
- [x] github.com/golang/groupcache
- [x] github.com/google/go-cmp
- [x] github.com/hashicorp/golang-lru
- [x] google.golang.org/genproto
- [x] ~~google.golang.org/grpc~~

sig-instrumentation:
- [x] github.com/prometheus/procfs (sig-instrumentation)
- [x] go.opencensus.io (sig-instrumentation)

Miscellaneous:
- [x] github.com/onsi/gomega (sig-testing)
- [x] github.com/vishvananda/netns (sig-network)
- [ ] golang.org/x/xerrors
- [x] google.golang.org/appengine (sig-auth)
- [ ] ~~github.com/cilium/ebpf~~
- [x] k8s.io/utils/mount (sig-storage)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: go.mod
Comment (liggitt) [inline go.mod]: is there a tagged level we can use?

File: go.mod
Comment (liggitt) [inline go.mod]: is there a tagged level we can use?

File: go.mod
Comment (dims) [reply go.mod]: asked them the question in https://github.com/microsoft/go-winio/issues/167

File: go.mod
Comment (dims) [reply go.mod]: cadvisor, we always cut a fresh release before our code freeze .. ( cc @dashpole ). Also may be better to wait until the go-winio gets sorted out first before we cut another cadvisor release

Comment (dashpole) [reply go.mod]: This is fine to do for now.  I'll update it to a tagged release before the next kubernetes release.

**Comment by BenTheElder** [comment]
> We use containerd 1.3.x in cluster-api / kind already, it's time to move up to newer dependencies.

we do but ... via CRI, I wouldn't expect any k/k repo changes to be related :O

**Comment by dims** [comment]
@BenTheElder right, i am just using that as an excuse to move the goal posts here :)

**Comment by dims** [comment]
@PatrickLang - can you please approve the dependency updates related to sig-windows? (cc @kubernetes/sig-windows-misc) 
@derekwaynecarr @mattjmcnaughton @Random-Liu - can you please ok the changes for sig-node? (cc @kubernetes/sig-node-pr-reviews)
@lavalamp @liggitt - can you please look at the ones from sig-apimachinery? (cc @kubernetes/sig-api-machinery-misc)
@spiffxp @stevekuznetsov - can you please look at the one for sig-testing? (cc @kubernetes/sig-testing)  
@thockin @dcbw - can you please look at the one for sig-network? (cc @kubernetes/sig-network-pr-reviews) 
@enj @mikedanese -  can you please look at the one for sig-auth? (cc @kubernetes/sig-auth-pr-reviews)

**Comment by dims** [comment]
@brancz @piosz can you please look at the dependencies for sig-instrumentation? (cc @kubernetes/sig-instrumentation-pr-reviews)

**Comment by dims** [comment]
/assign @thockin @smarterclayton 

(for root approval)

**Comment by dims** [comment]
/hold

(for reviews)

**Comment by lavalamp** [comment]
What do you think about updating gogoproto first in a separate change for easier testing & rollback?

**Comment by lavalamp** [comment]
lgtm for github.com/hashicorp/golang-lru

**Comment by mikedanese** [comment]
go-cmp and appengine lgtm

**Comment by mattjmcnaughton** [comment]
/uncc

going to let @derekwaynecarr or @dchen1107 review this for sig-node... i'm not sure i have the full context on all aspects of the `kubelet` to know what upgrading containerd could impact.

**Comment by liggitt** [comment]
github.com/golang/groupcache lgtm

**Review by stevekuznetsov — APPROVED** [review]
`vendor/github.com/onsi/gomega/` LGTM

File: go.mod
Comment (PatrickLang) [inline go.mod]: Based on this - we should still be using a tagged commit ID, not version. 0.8.7 wasn't tested with Kubernetes before it was tagged. If we want to use that as a testing point I'm ok with it, but I think we should tag the commit ID so the next person updating this doesn't assume there will be a relevant tagged version. 

https://github.com/microsoft/hcsshim/issues/642#issuecomment-571772207

Comment (dims) [reply go.mod]: @PatrickLang yes, let's please use it as a testing point.

**Comment by derekwaynecarr** [comment]
sig-node updates are fine.

/approve

File: go.mod
Comment (dims) [inline go.mod]: This change did not stick. 2 steps forward and one step back!

**Comment by dims** [comment]
/assign @liggitt 

Jordan,
this is ready. all green and got acks from all the sigs.

File: go.mod
Comment (liggitt) [inline go.mod]: this is a sunset repository used only by cadvisor redis storage. we should find a way to isolate storage and container integrations in cadvisor so we don't pull this into our dependency tree if we don't need to

File: go.mod
Comment (liggitt) [inline go.mod]: this pulls in a lot of additional transitive dependencies and is only used by cadvisor influxdb storage (which we don't use). we should find a way to isolate storage and container integrations in cadvisor so we don't pull this into our dependency tree if we don't need to

File: go.mod
Comment (liggitt) [inline go.mod]: undoes https://github.com/kubernetes/kubernetes/pull/84770

File: go.mod
Comment (liggitt) [inline go.mod]: this is only used by cadvisor elasticsearch storage (which we don't use). we should find a way to isolate storage and container integrations in cadvisor so we don't pull this into our dependency tree if we don't need to

File: go.mod
Comment (liggitt) [inline go.mod]: these pull in a lot of additional transitive dependencies and are only used by cadvisor kafka storage integration (which we don't use). we should find a way to isolate storage and container integrations in cadvisor so we don't pull this into our dependency tree if we don't need to

**Comment by liggitt** [comment]
@dims @dashpole since the last time we updated cadvisor, it added a go.mod file, which exposes all its transitive dependency version opinions, whether we make use of them or not. That means that all the container runtime (containerd, crio, docker, libcontainer, mesos, systemd) and storage integrations (bigquery, elasticsearch, influxdb, kafka, redis, statsd) that cadvisor has affect our dependency tree, even if we don't actually vendor the code in and link to it.

Unless there is an urgent reason to pull this in (e.g. fixing critical bugs), I'd like to see if we can isolate the cadvisor dependencies needed for the cadvisor cmd so downstream projects don't have to pull in the world like this. I'm experimenting with an approach locally in the cadvisor component and can sync up with you and @dashpole on it.

**Comment by dims** [comment]
@liggitt no rush other than the normal "try to get this stuff into master early in the cycle" so we can bake stuff in ci

**Comment by liggitt** [comment]
It looks like github.com/Microsoft/hcsshim@v0.8.7 also has a dependency on k8s.io/kubernetes@v1.13.0, which is problematic

https://github.com/microsoft/hcsshim/blob/master/go.mod#L36

File: go.mod
Comment (kazimsarikaya) [inline go.mod]: it fixes #87341

File: go.mod
Comment (kazimsarikaya) [inline go.mod]: it fixes #87341

File: go.mod
Comment (kazimsarikaya) [inline go.mod]: does #87341

File: go.mod
Comment (kazimsarikaya) [inline go.mod]: it fixes #87341

**Review by kazimsarikaya — APPROVED** [review]
zfs-go related problems (#87341 ) are solved with this pr

**Comment by liggitt** [comment]
> It looks like github.com/Microsoft/hcsshim@v0.8.7 also has a declared dependency on k8s.io/kubernetes@v1.13.0, which is problematic
> 
> https://github.com/microsoft/hcsshim/blob/master/go.mod#L36

That is depending on cri-api, which is published to a staging module now. https://github.com/liggitt/hcsshim/commits/bump-k8s easily updates hcsshim to use the staging module, but that would mean we could not depend on that version of hcsshim (we do not allow k8s.io/kubernetes -> vendor -> staging cycles). The only way I see to unblock that is for k8s.io/cri-api to graduate from staging and be a standalone repo.

**Comment by dims** [comment]
ah fun error

```
ERROR(linux/386,linux/arm) vendor/github.com/opencontainers/selinux/go-selinux/selinux_linux.go:118:17: unix.SELINUX_MAGIC (untyped int constant 4185718668) overflows int32
exit status 1
!!! Type Check has failed. This may cause cross platform build failures.
!!! Please see https://git.k8s.io/kubernetes/test/typecheck for more information.
```

File: go.mod
Comment (mattjmcnaughton) [inline go.mod]: Very possible I'm just misunderstanding how the release candidate tagging process works... but am I reading correctly that we're going from `v1.0.0-rc10` to `v1.0.0-rc8`? Would that not be a downgrade?

**Review by mattjmcnaughton — COMMENTED** [review]
W.r.t. `runc`, I think this looks good to me...

As far as I can tell, the main public facing impact is the addition of cgroups v2 support.

There is some deleted functionality, and some refactoring around where methods live... but that feels like implementation details.

I don't see any new `init` behavior or added dependencies or other things I'd consider suspicious.

File: go.mod
Comment (dims) [reply go.mod]: good point, will revise this to keep the rc10

**Comment by brancz** [comment]
cc @dashpole

**Comment by liggitt** [comment]
looks like something weird happened to the kube-proxy go.sum in the `github.com/checkpoint-restore/go-criu` and `github.com/cilium/ebpf` commits (added 445 lines, then dropped them)

**Comment by dims** [comment]
@liggitt when we get the cadvisor stuff merged, i can do this again without touching the go-criu version, so we should be fine. Let's work on getting the `isolate` branch merged there.

**Comment by dims** [comment]
This too a while to get right. Especially happy with the changes in cadvisor to minimize the transitive dependencies and hence the churn in our go.* files. We can move to current cadvisor as-is to get other things in (like `dockerless` tag) please @liggitt and @dashpole

**Comment by liggitt** [comment]
just a couple comments:
1. we're not pinned to a go1.13 branch of golang/x/net, so floating to the preferred version would make sense to me
2. the influxdb commit can be dropped, since we just undo that change in the "drop unused pinned versions" commit

I made both of those changes in https://github.com/liggitt/kubernetes/commits/update-hcsshim-v0.8.7-and-containerd-v1.3.2 if you want to pick those up

**Comment by dims** [comment]
@liggitt Done with both items you listed. yep, thanks for sharing your branch it was helpful.

**Comment by mattjmcnaughton** [comment]
/retest

Exciting stuff :) Looking forward to rebasing https://github.com/kubernetes/kubernetes/pull/87746 on this once its merged.

**Comment by liggitt** [comment]
/hold

For release-blocking utils bump, to avoid merge conflicts

**Comment by dims** [comment]
/hold cancel

#89444 is merged and the cherry-pick #89449 is merged as well.

**Comment by BenTheElder** [comment]
I don't know why yet but this seemingly broke kind clusters on certain hosts (google's desktop linux distro and docker for mac I think so far).

https://github.com/kubernetes-sigs/kind/issues/1569#issuecomment-624933076

Not exactly expecting a revert based on that but ... FYI. And interested in any thoughts why this would be the relevant commit. Possibly kubelet unhealthy + api server does not come up ... not sure why yet.

**Comment by liggitt** [comment]
nothing leaps to mind, though I'd check the kubelet logs for hints first, given it is responsible for running the apiserver static pod and is the component most affected by this commit. Mind opening an issue to track/discuss?

**Comment by dims** [comment]
fyi, this problem was already fixed in  https://github.com/google/cadvisor/pull/2471

**Comment by dghubble** [comment]
On Fedora CoreOS (but not Flatcar), Kubelet can't start static pods beginning here (in v1.19.0-alpha.0, specifically 4274ea2c89dee24e4c188a71e8164b2a40d1e181) and present in v1.19.0-beta.1. Filed as https://github.com/kubernetes/kubernetes/issues/91795, but wanted to provide a 2nd case to show its broader than just kind and google's OS in case that's helpful.

**Comment by BenTheElder** [comment]
this is still present in https://github.com/kubernetes/kubernetes/issues/89859#issuecomment-624963169 as well

thanks for the details!

**Comment by dims** [comment]
thanks @dghubble that's helpful.

**Pull request merged**
